runtime/builtin_types.go

```diff
@@ -190,6 +190,29 @@ func builtinAbs(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	return Abs(f, args[0])
 }
 
+func builtinAll(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	if raised := checkFunctionArgs(f, "all", args, ObjectType); raised != nil {
+		return nil, raised
+	}
+	iter, raised := Iter(f, args[0])
+	if raised != nil {
+		return nil, raised
+	}
+	o, raised := Next(f, iter)
+	for ; raised == nil; o, raised = Next(f, iter) {
+		if ret, raised := IsTrue(f, o); raised != nil {
+			return nil, raised
+		} else if !ret {
+			return False.ToObject(), nil
+		}
+	}
+	if !raised.isInstance(StopIterationType) {
+		return nil, raised
+	}
+	f.RestoreExc(nil, nil)
```

Comment:
It's unfortunate that there's so much boilerplate for iterating over a sequence. [seqForEach](https://github.com/google/grumpy/blob/master/runtime/seq.go#L128) almost does what you want but doesn't support breaking out of the loop early. We could add support for that by returning an additional bool value from the callback which signals whether to continue. Might be messy though. I'll leave it up to you whether you want to try something like that.